crates/wasm-mutate/src/mutators/codemotion/mutators/loop_unrolling.rs

```diff
@@ -0,0 +1,246 @@
+                            if *relative_depth > current_depth {
+                                // Out jump...annotate for fixing
+                                to_fix.insert(idx, Instruction::BrIf(relative_depth + 1));
+                            }
+                        }
+                        Operator::BrTable { table } => {
+                            let mut jmpfix = vec![];
+                            for i in table.targets() {
+                                let d = i?;
+                                if d > current_depth {
+                                    // Out jump...annotate for fixing
+                                    jmpfix.push(d + 1)
+                                } else {
+                                    jmpfix.push(d)
+                                }
+                            }
+
+                            let mut def = table.default();
+                            if def > current_depth {
+                                def += 1;
+                            }
+
+                            to_fix.insert(idx, Instruction::BrTable(jmpfix.into(), def));
+                        }
+                        _ => {}
+                    }
+                }
+
+                // println!("To fix {:?}", to_fix);
+                // Write the unroll
+                newfunc.instruction(&Instruction::Block(map_block_type(*ty)?));
+                // Write A' br B'
+                let including_chunk =
+                    operators[range.start + 1 /* skip the loop instruction */..range.end + 1]
+                        .iter();
+                self.write_and_fix_loop_body(including_chunk, &to_fix, newfunc, input_wasm)?;
+
+                // Write A' br B'
+                newfunc.instruction(&Instruction::Br(1));
+                newfunc.instruction(&Instruction::End);
+                // Write the Loop
+                newfunc.instruction(&Instruction::Loop(map_block_type(*ty)?));
+
+                // Write A' br B'
+                let including_chunk =
+                    operators[range.start + 1 /* skip the loop instruction */..range.end + 1]
+                        .iter();
+                self.write_and_fix_loop_body(including_chunk, &to_fix, newfunc, input_wasm)?;
+
+                // Closing loop
+                newfunc.instruction(&Instruction::End);
+
+                // Closing end
+                newfunc.instruction(&Instruction::End);
+            }
+            _ => unreachable!("Invalid node passed as a loop to unroll"),
+        }
+        Ok(())
+    }
+}
+
+impl AstWriter for LoopUnrollWriter {
+    fn write_loop<'a>(
+        &self,
+        ast: &Ast,
+        nodeidx: usize,
+        body: &[usize],
+        newfunc: &mut Function,
+        operators: &Vec<OperatorAndByteOffset>,
+        input_wasm: &'a [u8],
+        ty: &wasmparser::TypeOrFuncType,
+    ) -> crate::Result<()> {
+        if self.loop_to_mutate == nodeidx && !self.mutated.get() {
```

Comment:
Why do we need `self.mutated`? Is there a reason we can't rely on the node index alone?